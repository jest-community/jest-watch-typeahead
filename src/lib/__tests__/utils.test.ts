import chalk from 'chalk';
import {
  trimAndFormatPath,
  formatTestNameByPattern,
  highlight,
} from '../utils';

describe('trimAndFormatPath', () => {
  test.each`
    testPath                                           | pad  | columns
    ${'/project/src/gonna/fit/all.js'}                 | ${6} | ${80}
    ${'/project/src/trimmed_dir/foo.js'}               | ${6} | ${20}
    ${'/project/src/exactly/sep_and_basename.js'}      | ${6} | ${29}
    ${'/project/src/long_name_gonna_need_trimming.js'} | ${6} | ${40}
  `(
    'formats when testpath="$testPath", pad="$pad", and columns="$columns"',
    ({ testPath, pad, columns }) => {
      expect(
        trimAndFormatPath(pad, { rootDir: '/project' }, testPath, columns),
      ).toMatchSnapshot();
    },
  );
});

describe('formatTestNameByPattern', () => {
  test.each`
    testName           | pattern   | width
    ${'the test name'} | ${'the'}  | ${30}
    ${'the test name'} | ${'the'}  | ${25}
    ${'the test name'} | ${'the'}  | ${20}
    ${'the test name'} | ${'the'}  | ${15}
    ${'the test name'} | ${'the'}  | ${10}
    ${'the test name'} | ${'the'}  | ${5}
    ${'the test name'} | ${'test'} | ${30}
    ${'the test name'} | ${'test'} | ${25}
    ${'the test name'} | ${'test'} | ${20}
    ${'the test name'} | ${'test'} | ${15}
    ${'the test name'} | ${'test'} | ${10}
    ${'the test name'} | ${'test'} | ${5}
    ${'the test name'} | ${'name'} | ${30}
    ${'the test name'} | ${'name'} | ${25}
    ${'the test name'} | ${'name'} | ${20}
    ${'the test name'} | ${'name'} | ${15}
    ${'the test name'} | ${'name'} | ${10}
    ${'the test name'} | ${'name'} | ${5}
  `(
    'formats when testname="$testName", pattern="$pattern", and width="$width"',
    ({ testName, pattern, width }) => {
      expect(
        formatTestNameByPattern(testName, pattern, width),
      ).toMatchSnapshot();
    },
  );
});

describe('highlight', () => {
  const rawPath =
    '/Users/janedoe/my-project/src/__tests__/utils/experimentation/entry-point/parseEntryPoint.test.js';
  const pattern = 'parse';
  let filePath: string;

  /**
   * Helps us test that the highlight is placed correctly when `filePath`
   * contains ANSI characters from being formatted by chalk.
   * Passing a plain string for `filePath` results in false negatives.
   */
  const formatWithAnsi = (text: string) => chalk.dim(text);

  it('places highlight correctly when file path is not truncated', () => {
    filePath = formatWithAnsi(
      '__tests__/utils/experimentation/entry-point/parseEntryPoint.test.js',
    );

    expect(highlight(rawPath, filePath, pattern)).toMatchInlineSnapshot(
      `"<dim>__tests__/utils/experimentation/entry-point/</intensity></>parse</><dim>EntryPoint.test.js</intensity>"`,
    );
  });

  it('places highlight correctly when file path is truncated', () => {
    filePath = formatWithAnsi(
      '...tils/experimentation/entry-point/parseEntryPoint.test.js',
    );

    expect(highlight(rawPath, filePath, pattern)).toMatchInlineSnapshot(
      `"<dim>...tils/experimentation/entry-point/</intensity></>parse</><dim>EntryPoint.test.js</intensity>"`,
    );
  });

  it('places highlight correctly when file path has relative head', () => {
    filePath = formatWithAnsi(
      './src/__tests__/utils/experimentation/entry-point/parseEntryPoint.test.js',
    );

    expect(highlight(rawPath, filePath, pattern)).toMatchInlineSnapshot(
      `"<dim>./src/__tests__/utils/experimentation/entry-point/</intensity></>parse</><dim>EntryPoint.test.js</intensity>"`,
    );
  });
});
