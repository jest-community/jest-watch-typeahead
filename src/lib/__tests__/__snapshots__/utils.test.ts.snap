// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`formatTestNameByPattern formats when testname="the test name", pattern="name", and width="5" 1`] = `"</>...me</>"`;

exports[`formatTestNameByPattern formats when testname="the test name", pattern="name", and width="10" 1`] = `"<dim>...st </intensity></>name</>"`;

exports[`formatTestNameByPattern formats when testname="the test name", pattern="name", and width="15" 1`] = `"<dim>the test </intensity></>name</>"`;

exports[`formatTestNameByPattern formats when testname="the test name", pattern="name", and width="20" 1`] = `"<dim>the test </intensity></>name</>"`;

exports[`formatTestNameByPattern formats when testname="the test name", pattern="name", and width="25" 1`] = `"<dim>the test </intensity></>name</>"`;

exports[`formatTestNameByPattern formats when testname="the test name", pattern="name", and width="30" 1`] = `"<dim>the test </intensity></>name</>"`;

exports[`formatTestNameByPattern formats when testname="the test name", pattern="test", and width="5" 1`] = `"</>...</><dim>me</intensity>"`;

exports[`formatTestNameByPattern formats when testname="the test name", pattern="test", and width="10" 1`] = `"</>...st</><dim> name</intensity>"`;

exports[`formatTestNameByPattern formats when testname="the test name", pattern="test", and width="15" 1`] = `"<dim>the </intensity></>test</><dim> name</intensity>"`;

exports[`formatTestNameByPattern formats when testname="the test name", pattern="test", and width="20" 1`] = `"<dim>the </intensity></>test</><dim> name</intensity>"`;

exports[`formatTestNameByPattern formats when testname="the test name", pattern="test", and width="25" 1`] = `"<dim>the </intensity></>test</><dim> name</intensity>"`;

exports[`formatTestNameByPattern formats when testname="the test name", pattern="test", and width="30" 1`] = `"<dim>the </intensity></>test</><dim> name</intensity>"`;

exports[`formatTestNameByPattern formats when testname="the test name", pattern="the", and width="5" 1`] = `"</>...</><dim>me</intensity>"`;

exports[`formatTestNameByPattern formats when testname="the test name", pattern="the", and width="10" 1`] = `"</>...</><dim>st name</intensity>"`;

exports[`formatTestNameByPattern formats when testname="the test name", pattern="the", and width="15" 1`] = `"</>the</><dim> test name</intensity>"`;

exports[`formatTestNameByPattern formats when testname="the test name", pattern="the", and width="20" 1`] = `"</>the</><dim> test name</intensity>"`;

exports[`formatTestNameByPattern formats when testname="the test name", pattern="the", and width="25" 1`] = `"</>the</><dim> test name</intensity>"`;

exports[`formatTestNameByPattern formats when testname="the test name", pattern="the", and width="30" 1`] = `"</>the</><dim> test name</intensity>"`;

exports[`trimAndFormatPath formats when testpath="/project/src/exactly/sep_and_basename.js", pad="6", and columns="29" 1`] = `"<dim>.../</intensity><bold>sep_and_basename.js</intensity>"`;

exports[`trimAndFormatPath formats when testpath="/project/src/gonna/fit/all.js", pad="6", and columns="80" 1`] = `"<dim>src/gonna/fit/</intensity><bold>all.js</intensity>"`;

exports[`trimAndFormatPath formats when testpath="/project/src/long_name_gonna_need_trimming.js", pad="6", and columns="40" 1`] = `"<bold>...ong_name_gonna_need_trimming.js</intensity>"`;

exports[`trimAndFormatPath formats when testpath="/project/src/trimmed_dir/foo.js", pad="6", and columns="20" 1`] = `"<dim>..._dir/</intensity><bold>foo.js</intensity>"`;
